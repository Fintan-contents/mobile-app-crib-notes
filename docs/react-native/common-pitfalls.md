---
title: ハマりがちな落とし穴
---

React Nativeで開発する際に起きた困り事や不具合についてまとめています。
