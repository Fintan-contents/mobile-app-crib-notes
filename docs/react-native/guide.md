---
title: 何らかのガイドを書きたい気持ちはある
---
