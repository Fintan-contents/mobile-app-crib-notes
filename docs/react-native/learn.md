---
title: About
hide_table_of_contents: true
---

このコンテンツはReact Nativeでのモバイルアプリケーション開発を学習するためのコンテンツです。

ReactでWebアプリケーションを実装できるがReact Nativeでモバイルアプリケーションを実装したことがないアプリケーション開発者を対象として、次のことについて学ぶことを目的としています。

- ReactとReact Nativeの違い
- React Nativeで作成したモバイルアプリケーションの動作確認方法

また、作成したモバイルアプリケーションの配布方法やテスト手法についてはこのコンテンツでは触れません。

<!-- textlint-disable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Getting Started',
    to: '/react-native/learn/getting-started',
    summary: <>React Nativeでのアプリ開発に必要な環境を用意します。</>
  },
  {
    title: 'アプリの実装前に',
    to: '/react-native/learn/basic-concepts',
    summary: <>React Nativeでアプリを実装し始める前に最低限必要となる前提知識や、React NativeとReact Navigationの基本的な知識について学びます。</>
  },
  {
    title: 'ToDoアプリの実装',
    to: '/react-native/learn/todo-app',
    summary: <>React Nativeで簡単なToDoアプリを実装します。</>
  },
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->
