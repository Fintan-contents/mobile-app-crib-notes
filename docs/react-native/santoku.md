---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

React Nativeサンプルアプリケーションの設計ドキュメントです。

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
