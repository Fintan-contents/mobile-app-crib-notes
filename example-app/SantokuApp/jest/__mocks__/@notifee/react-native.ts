import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
