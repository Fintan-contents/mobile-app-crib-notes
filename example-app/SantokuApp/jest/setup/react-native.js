// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
