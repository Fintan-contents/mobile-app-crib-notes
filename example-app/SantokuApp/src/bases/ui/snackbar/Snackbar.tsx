import {m} from 'bases/message/Message';
import React, {useMemo, useState} from 'react';

import {
  SnackbarComponent,
  SnackbarHideProps as SnackbarComponentHideProps,
  SnackbarProps as SnackbarComponentProps,
  SnackbarShowProps as SnackbarComponentShowProps,
} from './SnackbarComponent';
import {FullWindowOverlay} from '../overlay/FullWindowOverlay';

type SnackbarShowProps = Omit<SnackbarComponentShowProps, 'message'>;
type SnackbarShowCloseButtonProps = Omit<SnackbarShowProps, 'actionText' | 'actionHandler' | 'actionTextStyle'>;
type SnackbarHideProps = Omit<SnackbarComponentHideProps, 'hide'>;

type SnackbarType = {
  /**
   * Show snackbar.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowContextProps).
   */
  show: (message: string, showProps?: SnackbarShowProps) => void;

  /**
   * Show the snackbar with the close button.
   * The close button is placed to the right of the message.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowCloseButtonContextProps).
   */
  showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => void;

  /**
   * Hide snackbar.
   *
   * @param hideProps - Snackbar props(SnackbarHideContextProps).
   */
  hide: (hideProps?: SnackbarHideProps) => void;
  Component: typeof Component;
};

let Snackbar: SnackbarType = {
  show: () => {
    throw new Error('Snackbar.Component is required.');
  },
  showWithCloseButton: () => {
    throw new Error('Snackbar.Component is required.');
  },
  hide: () => {
    throw new Error('Snackbar.Component is required.');
  },
  Component,
};

function Component(props: {initialState?: SnackbarComponentShowProps}) {
  const [state, setState] = useState<SnackbarComponentProps>(
    props.initialState ?? {
      message: '',
    },
  );
  // const snackbarContext = useMemo<SnackbarContextType>(
  Snackbar = useMemo<SnackbarType>(
    () => ({
      ...Snackbar,
      show: (message: string, showProps?: SnackbarShowProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          message,
        });
      },
      showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          actionText: m('閉じる'),
          actionHandler: Snackbar.hide,
          message,
        });
      },
      hide: (hideProps?: SnackbarHideProps) => {
        setState(prevState => ({...prevState, ...hideProps, hide: true}));
      },
    }),
    [],
  );
  return (
    <>
      <FullWindowOverlay>
        <SnackbarComponent {...state} />
      </FullWindowOverlay>
    </>
  );
}

export {Snackbar};
