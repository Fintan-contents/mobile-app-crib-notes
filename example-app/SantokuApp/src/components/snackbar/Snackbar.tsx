import React, {useCallback, useRef, useState} from 'react';
import {Animated, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, View, ViewStyle} from 'react-native';

import {FullWindowOverlay} from './FullWindowOverlay';

import CompositeAnimation = Animated.CompositeAnimation;

export type SnackbarShowProps = {
  message: string;
  messageTextStyle?: StyleProp<TextStyle>;
  top?: number;
  bottom?: number;
  right?: number;
  left?: number;
  style?: StyleProp<ViewStyle>;
  animatedContainerStyle?: StyleProp<ViewStyle>;
  actionText?: string;
  actionHandler?: () => void;
  actionTextStyle?: StyleProp<TextStyle>;
  autoHideDuration?: number;
  fadeInDuration?: number;
  fadeOutDuration?: number;
  forceFadeOutDuration?: number;
};

export type SnackbarHideProps = {
  hide?: true;
  hideFadeOutDuration?: number;
};

export type SnackbarProps = SnackbarShowProps & SnackbarHideProps;

export const Snackbar: React.FC<SnackbarProps> = (props) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const fadeInAnimationRef = useRef<CompositeAnimation>();
  const fadeOutAnimationRef = useRef<CompositeAnimation>();
  const barrierFadeOutAnimationRef = useRef<CompositeAnimation>();
  const [visibleSnackbarProps, setVisibleSnackbarProps] = useState<SnackbarShowProps>();

  const animationStart = useCallback(
    (
      animationRef: React.MutableRefObject<CompositeAnimation | undefined>,
      config: Animated.TimingAnimationConfig,
      callback?: Animated.EndCallback,
    ) => {
      const animation = Animated.timing(fadeAnim, config);
      animationRef.current = animation;
      animation.start((result) => {
        animationRef.current = undefined;
        if (callback) {
          callback(result);
        }
      });
    },
    [fadeAnim],
  );

  const show = useCallback(() => {
    setVisibleSnackbarProps(props);

    const fadeInAnimationConfig = {
      toValue: 1,
      duration: props.fadeInDuration,
      useNativeDriver: true,
    };
    animationStart(fadeInAnimationRef, fadeInAnimationConfig, ({finished}) => {
      if (finished) {
        const fadeOutAnimationConfig = {
          toValue: 0,
          delay: props.autoHideDuration,
          duration: props.fadeOutDuration,
          useNativeDriver: true,
        };
        animationStart(fadeOutAnimationRef, fadeOutAnimationConfig, () => {
          if (!barrierFadeOutAnimationRef) {
            setVisibleSnackbarProps(undefined);
          }
        });
      } else {
        if (!barrierFadeOutAnimationRef) {
          setVisibleSnackbarProps(undefined);
        }
      }
    });
  }, [props, animationStart]);

  const forceFadeout = useCallback(
    (fadeOutDuration?: number, callback?: () => void) => {
      const barrierFadeOutAnimationConfig = {
        toValue: 0,
        duration: fadeOutDuration,
        useNativeDriver: true,
      };

      fadeInAnimationRef.current?.stop();
      fadeOutAnimationRef.current?.stop();

      animationStart(barrierFadeOutAnimationRef, barrierFadeOutAnimationConfig, () => {
        setVisibleSnackbarProps(undefined);
        callback?.();
      });
    },
    [animationStart],
  );

  React.useEffect(() => {
    if (props.hide) {
      forceFadeout(props.hideFadeOutDuration ?? props.forceFadeOutDuration);
      return;
    }
    if (!props.message) {
      return;
    }
    if (barrierFadeOutAnimationRef.current) {
      return;
    }
    if (fadeInAnimationRef.current || fadeOutAnimationRef.current) {
      forceFadeout(props.forceFadeOutDuration, show);
      return;
    }
    show();
  }, [props, show, forceFadeout]);

  const snackbarStyle = StyleSheet.flatten([styles.snackbar, props.style]);

  const animatedViewStyle = StyleSheet.flatten<ViewStyle>([styles.animatedContainer, props.animatedContainerStyle]);
  if (animatedViewStyle?.top === undefined && animatedViewStyle.bottom === undefined) {
    animatedViewStyle.bottom = 20;
  }

  return (
    <>
      {props.children}
      {visibleSnackbarProps && (
        <FullWindowOverlay>
          <Animated.View style={StyleSheet.flatten([{opacity: fadeAnim}, animatedViewStyle])}>
            <View style={snackbarStyle}>
              <View style={styles.messageContainer}>
                <Text style={styles.messageText}>{visibleSnackbarProps.message}</Text>
              </View>
              {visibleSnackbarProps.actionText && visibleSnackbarProps.actionHandler && (
                <View style={styles.actionContainer}>
                  <TouchableOpacity onPress={visibleSnackbarProps.actionHandler}>
                    <Text style={styles.actionText}>{visibleSnackbarProps.actionText}</Text>
                  </TouchableOpacity>
                </View>
              )}
            </View>
          </Animated.View>
        </FullWindowOverlay>
      )}
    </>
  );
};

Snackbar.defaultProps = {
  autoHideDuration: 4000,
  fadeInDuration: 1000,
  fadeOutDuration: 1000,
  forceFadeOutDuration: 300,
};

const styles = StyleSheet.create({
  animatedContainer: {
    flex: 1,
    position: 'absolute',
    minHeight: 48,
    paddingHorizontal: 10,
    width: '100%',
  },
  snackbar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    backgroundColor: '#393939',
  },
  messageContainer: {
    flex: 1,
    paddingVertical: 20,
    paddingLeft: 20,
    paddingRight: 10,
  },
  actionContainer: {
    flex: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingVertical: 20,
    paddingLeft: 10,
    paddingRight: 20,
  },
  messageText: {
    color: 'white',
  },
  actionText: {
    color: '#85A3F1',
  },
});
