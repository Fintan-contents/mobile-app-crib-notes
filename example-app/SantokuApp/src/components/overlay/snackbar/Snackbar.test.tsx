import {act} from '@testing-library/react-hooks';
import {render, screen, waitFor} from '@testing-library/react-native';
import React from 'react';
import {Text, TextStyle, ViewStyle} from 'react-native';
import {ReactTestInstance} from 'react-test-renderer';

import {Snackbar} from './Snackbar';

jest.useFakeTimers();

function getStyle<T>(instance: ReactTestInstance) {
  return instance.props.style as T;
}

const ChildComponent = () => {
  return <Text testID="text">test</Text>;
};

const AUTO_HIDE_DURATION = 4000;
const FADE_IN_DURATION = 1000;
const FADE_OUT_DURATION = 1000;
const FORCE_FADE_OUT_DURATION = 300;
const HIDE_FADE_OUT_DURATION = 300;

describe('Snackbar', () => {
  it('Snackbarが正常にrenderできることを確認', () => {
    render(
      <Snackbar message="テストメッセージ">
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByTestId('text')).not.toBeNull();
    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).opacity).toBe(0);
    expect(screen).toMatchSnapshot('render直後');

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION);
    });
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).opacity).toBe(1);
    expect(screen).toMatchSnapshot('フェードイン後');

    act(() => {
      jest.advanceTimersByTime(AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });
    expect(screen.queryByTestId('snackbarAnimatedView')).toBeNull();
    expect(screen).toMatchSnapshot('フェードアウト後');
  });

  it('Snackbar表示中にpropsが更新された場合、フェードアウト後に更新後のpropsでSnackbarが表示されることを確認', async () => {
    render(
      <Snackbar message="初回">
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('初回')).not.toBeNull();

    screen.update(
      <Snackbar message="２回目">
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('初回')).not.toBeNull();

    await waitFor(() => {
      jest.advanceTimersByTime(FORCE_FADE_OUT_DURATION);

      expect(screen.queryByText('初回')).toBeNull();
      expect(screen.queryByText('２回目')).not.toBeNull();
    });
  });

  it('Snackbarの表示中に２連続propsが更新された場合、後で更新した方のpropsでSnackbarが表示されることを確認', async () => {
    render(
      <Snackbar message="初回">
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('初回')).not.toBeNull();

    screen.update(
      <Snackbar message="２回目">
        <ChildComponent />
      </Snackbar>,
    );

    screen.update(
      <Snackbar message="３回目">
        <ChildComponent />
      </Snackbar>,
    );

    await waitFor(() => {
      jest.advanceTimersByTime(FORCE_FADE_OUT_DURATION);

      expect(screen.queryByText('３回目')).not.toBeNull();
    });
  });

  it('Snackbarの表示後に同一のpropsを指定した場合、Snackbarが表示されないことを確認', () => {
    const props = {
      message: 'テストメッセージ',
      messageTextStyle: {color: 'white'},
      style: {backgroundColor: 'aqua'},
      positionStyle: {minHeight: 60},
      actionText: 'close',
      actionHandler: jest.fn(),
      actionTextStyle: {color: 'red'},
      autoHideDuration: 100,
      fadeInDuration: 200,
      fadeOutDuration: 300,
      forceFadeOutDuration: 400,
      timestamp: Date.now(),
    };
    render(
      <Snackbar {...props}>
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION + AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });

    screen.update(
      <Snackbar {...props}>
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('テストメッセージ')).toBeNull();
  });

  it('Snackbarの表示後にTimestamp以外同一のpropsを指定した場合、Snackbarが表示されることを確認', () => {
    const props = {
      message: 'テストメッセージ',
      messageTextStyle: {color: 'white'},
      style: {backgroundColor: 'aqua'},
      positionStyle: {minHeight: 60},
      actionText: 'close',
      actionHandler: jest.fn(),
      actionTextStyle: {color: 'red'},
      autoHideDuration: 100,
      fadeInDuration: 200,
      fadeOutDuration: 300,
      forceFadeOutDuration: 400,
    };
    render(
      <Snackbar {...props} timestamp={Date.now()}>
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION + AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });

    screen.update(
      <Snackbar {...props} timestamp={Date.now()}>
        <ChildComponent />
      </Snackbar>,
    );

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
  });

  it('Snackbar表示中にpropsでhideを指定した場合、Snackbarが消えることを確認', async () => {
    render(
      <Snackbar message="テストメッセージ">
        <ChildComponent />
      </Snackbar>,
    );

    jest.advanceTimersByTime(FADE_IN_DURATION);
    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    screen.update(
      <Snackbar message="テストメッセージ" hide>
        <ChildComponent />
      </Snackbar>,
    );

    await waitFor(() => {
      jest.advanceTimersByTime(HIDE_FADE_OUT_DURATION);
      expect(screen.queryByText('テストメッセージ')).toBeNull();
    });
  });

  it('Snackbarに指定したpropsがrenderに反映されていることを確認', () => {
    /**
     * 下記項目は対象外
     * - autoHideDuration / fadeInDuration / fadeOutDuration / forceFadeOutDuration / hideFadeOutDuration
     *     テストコードでアニメーション時間を調整できない
     * - actionHandler
     *     setVisibleSnackbarProps()でpropsをセットした時点で、指定したモック関数ではなくなっており、追跡できない
     * - hide
     *     テスト済
     */
    render(
      <Snackbar
        message="テストメッセージ"
        messageTextStyle={{color: 'black'}}
        style={{backgroundColor: 'red'}}
        positionStyle={{bottom: 50}}
        actionText="閉じる"
        actionHandler={() => {}}
        actionTextStyle={{color: 'blue'}}>
        <ChildComponent />
      </Snackbar>,
    );

    jest.advanceTimersByTime(FADE_IN_DURATION);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('テストメッセージ')).color).toBe('black');
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarStyleView')).backgroundColor).toBe('red');
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).bottom).toBe(50);
    expect(screen.queryByText('閉じる')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('閉じる')).color).toBe('blue');
  });
});
