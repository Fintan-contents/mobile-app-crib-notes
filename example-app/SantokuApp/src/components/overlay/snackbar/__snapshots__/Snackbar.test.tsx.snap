// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Snackbarが正常にrenderできることを確認: render直後 1`] = `
Array [
  <Text
    testID="text"
  >
    test
  </Text>,
  <View
    testID="FullWindowOverlay"
  >
    <View
      collapsable={false}
      style={
        Object {
          "bottom": 20,
          "flex": 1,
          "minHeight": 48,
          "opacity": 0,
          "paddingHorizontal": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="snackbarAnimatedView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="snackbarStyleView"
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            テストメッセージ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`Snackbar Snackbarが正常にrenderできることを確認: フェードアウト後 1`] = `
<Text
  testID="text"
>
  test
</Text>
`;

exports[`Snackbar Snackbarが正常にrenderできることを確認: フェードイン後 1`] = `
Array [
  <Text
    testID="text"
  >
    test
  </Text>,
  <View
    testID="FullWindowOverlay"
  >
    <View
      collapsable={false}
      style={
        Object {
          "bottom": 20,
          "flex": 1,
          "minHeight": 48,
          "opacity": 1,
          "paddingHorizontal": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="snackbarAnimatedView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="snackbarStyleView"
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            テストメッセージ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
